### ISSUE TITLE
rake paperclip:refresh CLASS=Post problem

### ISSUE THREAD

**Issue by adolfopeccin** [thoughtbot/paperclip#1748]
Hello,
I'm running `rake paperclip:refresh CLASS=Post` to regenerate images on S3 but it keep generating 0 bytes images inside originals folder on S3 and no fix for broken images... logs i have like this:

```
I, [2015-01-21T23:27:59.971241 #4990]  INFO -- : [paperclip] copying posts/4/original/ae776ce234ec08ecd4e1415a18f38c3891aa502a.JPG to local file /tmp/ece8085f41a53dd109d55e1fd880223c20150121-4990-tp5jz8.JPG
I, [2015-01-21T23:27:59.990304 #4990]  INFO -- : [AWS S3 200 0.017615 0 retries] get_object(:bucket_name=>"content.mysite.com",:key=>"posts/4/original/ae776ce234ec08ecd4e1415a18f38c3891aa502a.JPG")  

I, [2015-01-21T23:27:59.992431 #4990]  INFO -- : Command :: file -b --mime '/tmp/ece8085f41a53dd109d55e1fd880223c20150121-4990-jz2i3f.JPG'
I, [2015-01-21T23:27:59.996744 #4990]  INFO -- : [paperclip] Content Type Spoof: Filename quadro-sloth.JPG (["image/jpeg"]), content type discovered from file command: inode/x-empty. See documentation to allow this combination.
I, [2015-01-21T23:27:59.998547 #4990]  INFO -- : [paperclip] saving posts/4/original/795c168900b2605500be18a812ee7c7323e5623a.JPG
I, [2015-01-21T23:28:00.025893 #4990]  INFO -- : [AWS S3 200 0.026421 0 retries] put_object(:acl=>:public_read,:bucket_name=>"content.mysite.com",:content_length=>0,:content_type=>"image/jpeg",:data=>Paperclip::AttachmentAdapter: quadro-sloth.JPG,:key=>"posts/4/original/795c168900b2605500be18a812ee7c7323e5623a.JPG")  
```

I dont know why i cant fix the broken images...

**Comment by reshleman** [comment]
I suspect this is related to #1729 and an issue with the combination of `paperclip` (4.2.1) and `aws-sdk` (1.5.8).

For me, running `rake paperclip:refresh:missing_styles` using this combination of gem versions resulted in 0-byte images being stored in the `original` folder on S3 and Paperclip storing a reference to the new, 0-byte version in the database.

(FWIW, I'm using fingerprinted filenames, and the original version was not deleted from the `original` folder when the new, 0-byte file was created.)

After updating `aws-sdk` to 1.64.0 (per #1729), running `rake paperclip:refresh:missing_styles` still fails with the same `Content Type Spoof` error, since Paperclip has already stored a reference to the newer (0-byte) version of the files. The task attempts to regenerate styles from the 0-byte "originals", and thus fails.

(I *can* verify that, after updating `aws-sdk`, 0-byte files are no longer generated when running the rake task *for new, non-empty files that are correctly referenced in the database*. That is, updating the gem prevents this issue from occurring, but does not resolve the issue for users who have already generated 0-byte originals.)

I'm not sure what the solution is here. Fortunately, I encountered this on a staging server, so it's not a huge issue. One possible workaround would be to rename the existing `original

 (I verified this by re-naming an old original to match the new 0-byte version name stored by Paperclip, then running the task. New styles were correctly generated for this image.)

**Comment by reshleman** [comment]
I've opened #1838 w/ a documentation update to prevent users from encountering this issue in the future.

**Comment by vladimir-e** [comment]
Thanks @reshleman your post is a great help!

**Comment by croaky** [comment]
I've been seeing the `Content Type Spoof`0-byte S3 issue with `aws-sdk` 1.64.0, `paperclip` 4.3.0

**Comment by tute** [comment]
Paperclip now only supports AWS SDK v2 or greater. Is this still an issue? Thanks!

**Comment by reshleman** [comment]
@tute I didn't open this, but I believe this was resolved by updating to `aws-sdk` `~> 1.6`. Presumably it's no longer an issue with v2. (Don't currently have an app handy to test on, unfortunately.)

**Comment by tute** [comment]
Thanks!
